src/test/java/com/connexta/ingest/adaptors/S3MetacardAdaptorTests.java

```diff
@@ -0,0 +1,153 @@
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class S3MetacardAdaptorTests {
+
+  private static final String MINIO_ADMIN_ACCESS_KEY = "admin";
+  private static final String MINIO_ADMIN_SECRET_KEY = "12345678";
+  private static final int MINIO_PORT = 9000;
+  private static AmazonS3Configuration configuration;
+  private static AmazonS3 amazonS3;
+  private static MetacardStorageAdaptor storageAdaptor;
+  private static String BUCKET = "metacard-quarantine";
+
+  @Container
+  public static final GenericContainer minioContainer =
+      new GenericContainer("minio/minio:RELEASE.2019-07-10T00-34-56Z")
+          .withEnv("MINIO_ACCESS_KEY", MINIO_ADMIN_ACCESS_KEY)
+          .withEnv("MINIO_SECRET_KEY", MINIO_ADMIN_SECRET_KEY)
+          .withExposedPorts(MINIO_PORT)
+          .withCommand("server --compat /data/metacard-quarantine")
+          .waitingFor(
+              new HttpWaitStrategy()
+                  .forPath("/minio/health/ready")
+                  .withStartupTimeout(Duration.ofSeconds(30)));
+
+  private String metacardContents;
+
+  @BeforeAll
+  public static void setUp() {
+    configuration =
+        new AmazonS3Configuration(
+            String.format(
+                "http://%s:%d",
+                minioContainer.getContainerIpAddress(), minioContainer.getMappedPort(MINIO_PORT)),
+            "local",
+            MINIO_ADMIN_ACCESS_KEY,
+            MINIO_ADMIN_SECRET_KEY);
+
+    amazonS3 =
+        AmazonS3ClientBuilder.standard()
+            .withEndpointConfiguration(
+                new AwsClientBuilder.EndpointConfiguration(
+                    configuration.getEndpoint(), configuration.getRegion()))
+            .withCredentials(
+                new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(
+                        configuration.getAccessKey(), configuration.getSecretKey())))
+            .enablePathStyleAccess()
+            .build();
+
+    storageAdaptor = new S3MetacardStorageAdaptor(amazonS3, BUCKET);
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    amazonS3.createBucket(BUCKET);
+  }
+
+  @AfterEach
+  public void afterEach() {
+    // Clear out S3
+    amazonS3.listObjects(BUCKET).getObjectSummaries().stream()
+        .map(S3ObjectSummary::getKey)
+        .forEach(key -> amazonS3.deleteObject(BUCKET, key));
+    amazonS3.deleteBucket(BUCKET);
+  }
+
+  @Test
+  public void testSuccessfulStoreRequest() {
+    storageAdaptor.store(4L, new ByteArrayInputStream("asdf".getBytes()), "1234");
+  }
+
+  @Test
+  public void testSuccessfulRetrieveRequest() {
+    final String key = "1234";
+    final String metacardContents = "asdf";
+    storageAdaptor.store(4L, new ByteArrayInputStream(metacardContents.getBytes()), key);
+    assertThat(
+        storageAdaptor.retrieve(key),
+        new TypeSafeMatcher<InputStream>() {
```

Comment:
:pencil2: Don't define new classes inline. Extract the definition of the new classes to a static inner class or its own file.